crates/node/benches/revm_spec.rs

```diff
@@ -0,0 +1,180 @@
+    } else if chain_spec.fork(EthereumHardfork::Berlin).active_at_head(block) {
+        reth_revm::primitives::BERLIN
+    } else if chain_spec.fork(EthereumHardfork::Istanbul).active_at_head(block) {
+        reth_revm::primitives::ISTANBUL
+    } else if chain_spec.fork(EthereumHardfork::Petersburg).active_at_head(block) {
+        reth_revm::primitives::PETERSBURG
+    } else if chain_spec.fork(EthereumHardfork::Byzantium).active_at_head(block) {
+        reth_revm::primitives::BYZANTIUM
+    } else if chain_spec.fork(EthereumHardfork::SpuriousDragon).active_at_head(block) {
+        reth_revm::primitives::SPURIOUS_DRAGON
+    } else if chain_spec.fork(EthereumHardfork::Tangerine).active_at_head(block) {
+        reth_revm::primitives::TANGERINE
+    } else if chain_spec.fork(EthereumHardfork::Homestead).active_at_head(block) {
+        reth_revm::primitives::HOMESTEAD
+    } else if chain_spec.fork(EthereumHardfork::Frontier).active_at_head(block) {
+        reth_revm::primitives::FRONTIER
+    } else {
+        panic!(
+            "invalid hardfork chainspec: expected at least one hardfork, got {:?}",
+            chain_spec.hardforks
+        )
+    }
+}
+
+
+fn revm_spec_two(chain_spec: &ChainSpec, block: &Head) -> SpecId {
+    enum Hardfork {
+        Ethereum(EthereumHardfork),
+        Optimism(OptimismHardfork),
+    }
+
+    static HARDFORKS: [(Hardfork, SpecId); 20] =  [
+        (Hardfork::Ethereum(EthereumHardfork::Prague), SpecId::PRAGUE_EOF),
+        (Hardfork::Optimism(OptimismHardfork::Granite), SpecId::GRANITE),
+        (Hardfork::Optimism(OptimismHardfork::Fjord), SpecId::FJORD),
+        (Hardfork::Optimism(OptimismHardfork::Ecotone), SpecId::ECOTONE),
+        (Hardfork::Optimism(OptimismHardfork::Canyon), SpecId::CANYON),
+        (Hardfork::Optimism(OptimismHardfork::Regolith), SpecId::REGOLITH),
+        (Hardfork::Optimism(OptimismHardfork::Bedrock), SpecId::BEDROCK),
+        (Hardfork::Ethereum(EthereumHardfork::Prague), SpecId::PRAGUE),
+        (Hardfork::Ethereum(EthereumHardfork::Cancun), SpecId::CANCUN),
+        (Hardfork::Ethereum(EthereumHardfork::Shanghai), SpecId::SHANGHAI),
+        (Hardfork::Ethereum(EthereumHardfork::Paris), SpecId::MERGE),
+        (Hardfork::Ethereum(EthereumHardfork::London), SpecId::LONDON),
+        (Hardfork::Ethereum(EthereumHardfork::Berlin), SpecId::BERLIN),
+        (Hardfork::Ethereum(EthereumHardfork::Istanbul), SpecId::ISTANBUL),
+        (Hardfork::Ethereum(EthereumHardfork::Petersburg), SpecId::PETERSBURG),
+        (Hardfork::Ethereum(EthereumHardfork::Byzantium), SpecId::BYZANTIUM),
+        (Hardfork::Ethereum(EthereumHardfork::SpuriousDragon), SpecId::SPURIOUS_DRAGON),
+        (Hardfork::Ethereum(EthereumHardfork::Tangerine), SpecId::TANGERINE),
+        (Hardfork::Ethereum(EthereumHardfork::Homestead), SpecId::HOMESTEAD),
+        (Hardfork::Ethereum(EthereumHardfork::Frontier), SpecId::FRONTIER),
+    ];
+
+    let hardforks = HARDFORKS.iter().rev();
```

Comment:
this makes it always makes the loop iterate only once since frontier is active in all cases